File: shop/src/main/java/com/coder/shop/services/UserService.java
package com.coder.shop.services;

import java.util.List;
import java.util.Set;

import org.springframework.stereotype.Service;

import com.coder.shop.models.AppUser;
import com.coder.shop.models.Role;

@Service
public interface UserService {
    List<AppUser> all();
    void register(AppUser user);
    void addRole(Long userId, int roleId);
    void removeRole(Long userId, int roleId);
    AppUser get(Long id);
    public Set<Role> getUserRoles(Long userId);
}

File: shoppy/src/app/users/userservice.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import Vary from '../utils/Vary';
import { UserModel } from '../models/usermodel';

@Injectable({
  providedIn: 'root',
})
export class Userservice {
  private readonly baseUrl = Vary.USERS_URL;
  constructor(private http: HttpClient) {}

  get(id: number) {
    return this.http.get<UserModel>(this.baseUrl + '/' + id);
  }
  all() {
    return this.http.get<UserModel[]>(this.baseUrl);
  }
  addRole(userId: number, roleId: number) {
    return this.http.patch<any>(this.baseUrl + '/add/role/' + userId + '/' + roleId, {});
  }
  removeRole(userId: number, roleId: number) {
    return this.http.patch<any>(this.baseUrl + '/remove/role/' + userId + '/' + roleId, {});
  }
}
